src/test/java/org/gridsuite/odre/server/OdreControllerTest.java

```diff
@@ -44,12 +50,38 @@ public void setUp() {
 
     @Test
     public void test() throws Exception {
+        byte[] aerialLinesBytes = IOUtils.toByteArray(new FileInputStream(ResourceUtils.getFile("classpath:lignes-aeriennes-rte.csv")));
+        byte[] undergroundLinesBytes = IOUtils.toByteArray(new FileInputStream(ResourceUtils.getFile("classpath:lignes-souterraines-rte.csv")));
+        byte[] substationsBytes = IOUtils.toByteArray(new FileInputStream(ResourceUtils.getFile("classpath:postes-electriques-rte.csv")));
+
+        MockMultipartFile file = new MockMultipartFile("file", "postes-electriques-rte.csv", "text/csv", substationsBytes);
+        MockMultipartFile substationsFile = new MockMultipartFile("files", "postes-electriques-rte.csv", "text/csv", substationsBytes);
+        MockMultipartFile aerialLinesFile = new MockMultipartFile("files", "lignes-aeriennes-rte.csv", "text/csv", aerialLinesBytes);
+        MockMultipartFile undergroundLinesFile = new MockMultipartFile("files", "lignes-souterraines-rte.csv", "text/csv", undergroundLinesBytes);
+        MockMultipartFile fileWithInvalidName = new MockMultipartFile("file", "postes-electriques.csv", "text/csv", substationsBytes);
+        MockMultipartFile substationsFileWithInvalidName = new MockMultipartFile("files", "postes-electriques.csv", "text/csv", substationsBytes);
+        MockMultipartFile aerialLinesFileWithInvalidName = new MockMultipartFile("files", "lignes-aeriennes.csv", "text/csv", aerialLinesBytes);
+        MockMultipartFile undergroundLinesFileWithInvalidName = new MockMultipartFile("files", "lignes-souterraines.csv", "text/csv", undergroundLinesBytes);
+
         mvc.perform(post("/" + OdreController.API_VERSION + "/substations")
                 .contentType(APPLICATION_JSON))
                 .andExpect(status().isOk());
 
         mvc.perform(post("/" + OdreController.API_VERSION + "/lines")
                 .contentType(APPLICATION_JSON))
                 .andExpect(status().isOk());
+
+        mvc.perform(multipart("/" + OdreController.API_VERSION + "/substations/upload").file(file))
+                .andExpect(status().isOk());
+
+        mvc.perform(multipart("/" + OdreController.API_VERSION + "/lines/upload").file(substationsFile).file(aerialLinesFile).file(undergroundLinesFile))
+                .andExpect(status().isOk());
+
+        // test with invalid file name
+        mvc.perform(multipart("/" + OdreController.API_VERSION + "/substations/upload").file(fileWithInvalidName))
+                .andExpect(status().isBadRequest());
+        // test with invalid file name
+        mvc.perform(multipart("/" + OdreController.API_VERSION + "/lines/upload").file(substationsFileWithInvalidName).file(aerialLinesFileWithInvalidName).file(undergroundLinesFileWithInvalidName))
+                .andExpect(status().isBadRequest());
```

Comment:
We could add more tests, it doesn't cost a lot, for example add a test where only one file has an invalidName

Reply:
add a case where we send more than 3 files well named
add a case where we send only 2 files well named

Reply:
Add a test for has CSV format maybe ? (for coverage)